pkg/k8s/k8s.go

```diff
@@ -471,15 +472,60 @@ func (c *cluster) CreateClusterBom(ctx context.Context) (*bom.Result, error) {
 	return c.getClusterBomInfo(components, nodesInfo)
 }
 
-func GetContainer(hex string, imageName containerimage.Reference) (bom.Container, error) {
-	repoName := imageName.Context().RepositoryStr()
-	registryName := imageName.Context().RegistryStr()
+func extractTag(imageName, digest string) string {
+	if !strings.Contains(imageName, ":") {
+		return containerimage.DefaultTag
+	}
+	parts := strings.Split(imageName, ":")
+	return strings.TrimSuffix(parts[len(parts)-1], "@"+digest)
+}
+
+func extractDigest(imageId string) string {
+	if strings.Contains(imageId, "@") {
+		imageId = strings.Split(imageId, "@")[1]
+	}
+	return strings.TrimPrefix(imageId, string(digest.Canonical)+":")
+
+}
+
+// GetContainer returns a container object based on `imageName` (pod.Spec.Containers.Name)
+// and `imageId` (pod.Status.ContainerStatuses.ImageID).
+func GetContainer(imageName, imageId string) (bom.Container, error) {
+	if strings.Contains(imageName, "@") {
+		imageName = strings.Split(imageName, "@")[0]
+	}
+	imageRef, err := utils.ParseReference(imageName)
+	if err != nil {
+		return bom.Container{}, fmt.Errorf("unable to parse imageRef name reference %q: %v", imageName, err)
+	}
+	// parse imageId to get the digest
+	hex := extractDigest(imageId)
+	// skip non sha256 digests
+	if len(hex) != digest.Canonical.Size()*2 {
+		return bom.Container{}, fmt.Errorf("unable to parse digest %q for %q", imageId, imageName)
```

Comment:
ditto, can we add an unhappy path test case for this?

Reply:
covered here https://github.com/aquasecurity/trivy-kubernetes/pull/452/commits/028d56ae5600e349d1736ebc6328e2b7ca4bf123

Reply:
added https://github.com/aquasecurity/trivy-kubernetes/pull/452/commits/66be7bab0cccc3e6053ae4d7da77e1d1c1555425